mir_perception/mir_handle_detection/ros/src/drawer_handle_perceiver.cpp

```diff
@@ -0,0 +1,256 @@
+#include <mir_handle_detection/drawer_handle_perceiver.h>
+
+DrawerHandlePerceiver::DrawerHandlePerceiver() : nh("~")
+{
+    nh.param<std::string>("output_frame", this->output_frame, "base_link_static");
+    nh.param<bool>("enable_debug_pc_pub", this->enable_debug_pc_pub, "true");
+
+    this->pc_sub = nh.subscribe("input_point_cloud", 1, &DrawerHandlePerceiver::pcCallback, this);
+    this->pose_pub = nh.advertise<geometry_msgs::PoseStamped>("output_pose", 1);
+    this->event_in_sub = nh.subscribe("event_in", 1, &DrawerHandlePerceiver::eventInCallback, this);
+    this->event_out_pub = nh.advertise<std_msgs::String>("event_out", 1);
+
+    if (this->enable_debug_pc_pub)
+        this->pc_pub = nh.advertise<sensor_msgs::PointCloud2>("output_point_cloud", 1);
+
+    float z_threshold_max, z_threshold_min;
+    nh.param<float>("z_threshold_min", z_threshold_min, -0.1);
+    nh.param<float>("z_threshold_max", z_threshold_max, 0.3);
+    this->passthrough_filter_z.setFilterFieldName ("z");
+    this->passthrough_filter_z.setFilterLimits (z_threshold_min, z_threshold_max);
+
+    float y_threshold_max, y_threshold_min;
+    nh.param<float>("y_threshold_min", y_threshold_min, -0.1);
+    nh.param<float>("y_threshold_max", y_threshold_max, 0.1);
+    this->passthrough_filter_y.setFilterFieldName ("y");
+    this->passthrough_filter_y.setFilterLimits (y_threshold_min, y_threshold_max);
+
+    float leaf_size_x, leaf_size_y, leaf_size_z;
+    nh.param<float>("leaf_size_x", leaf_size_x, 0.01);
+    nh.param<float>("leaf_size_y", leaf_size_y, 0.01);
+    nh.param<float>("leaf_size_z", leaf_size_z, 0.01);
+    this->voxel_grid_filter.setLeafSize (leaf_size_x, leaf_size_y, leaf_size_z);
+
+    float seg_dist_threshold;
+    nh.param<float>("seg_dist_threshold", seg_dist_threshold, 0.01);
+    this->seg.setOptimizeCoefficients(true);
+    this->seg.setModelType(pcl::SACMODEL_PLANE);
```

Comment:
this should be `SACMODEL_NORMAL_PARALLEL_PLANE` with setAxis with axis = [1.0, 0.0, 0.0]